src/util/simplify_expr_pointer.cpp

```diff
@@ -417,39 +418,67 @@ bool simplify_exprt::simplify_pointer_offset(exprt &expr)
   return true;
 }
 
+static exprt simplify_inequality_address_of_prepare_op(const exprt &op)
+{
+  exprt result = op;
```

Comment:
Think I commented on the original, but a `PRECONDITION(op.typ()==ID_pointer)` I suspect might make the parameter to this function a bit more obvious what it is meant to be. Also consider giving it a more descriptive name (e.g. `pointer_expression`)